core/src/scene/sceneLoader.cpp

```diff
@@ -896,11 +899,34 @@ void SceneLoader::loadSource(const std::string& name, const Node& source, const
     }
 
     std::string type = source["type"].Scalar();
-    std::string url = source["url"].Scalar();
+    std::string url;
+    std::string mbtiles;
     int32_t minDisplayZoom = -1;
     int32_t maxDisplayZoom = -1;
     int32_t maxZoom = 18;
 
+    std::string mime;
+
+    if (type == "GeoJSON") {
```

Comment:
yes, this should be improved - I was not sure which way to go: when passing `mime` to MBTilesDataSource constructor `*Source::getMime(string url)` must be static.